components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -421,40 +470,47 @@ func (g *rtkSerial) receiveAndWriteSerial() {
 		msg, err := scanner.NextMessage()
 		if err != nil {
 			g.ntripMu.Lock()
-			g.ntripStatus = false
+			g.connectedToNtrip = false
 			g.ntripMu.Unlock()
 
 			if msg == nil {
 				if g.isClosed {
 					return
 				}
+
 				g.logger.Debug("No message... reconnecting to stream...")
-				err = g.getStream(g.ntripClient.MountPoint, g.ntripClient.MaxConnectAttempts)
-				if err != nil {
-					g.err.Set(err)
-					return
+				if g.isVirtualBase {
+					g.rw, err = g.sendGGAMessage()
+					if err != nil && !errors.Is(err, io.EOF) {
+						g.err.Set(err)
+					}
+					g.rw, err = g.sendGGAMessage()
```

Comment:
Can use some advice on how to properly handle this. When we get an EOF error, we must call sendGGAMessage once more to redial the port. But this doesn't look like a good way to approach it.

Reply:
have an error bool intialized to some value at the beginning of trying to write the gga messages to the virtual base that turns into true or false upon a successful attempt?

Reply:
I kinda understand, because in line 486 she'c checking the error and comparing is to EOF, and setting it only if it is not EOF, so at some point we're expecting it not to be EOF - we reconnected to the serial port, the baudrate was set correctly).

One thing is - this spams logs if it is never not EOF, so we should ensure that there is a stopping point in separate work.